### PR TITLE
Enable Orbital Optimization  for hybridRep orbitals

### PR DESCRIPTION
**Pull request by camelto2**
Please review the [developer documentation](https://github.com/QMCPACK/qmcpack/wiki/Development-workflow)
on the wiki of this project that contains help and requirements.

## Proposed changes

This PR enables OO for hybrid rep orbitals. 

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- New feature

### Does this introduce a breaking change?
- No

## What systems has this change been tested on?
M1 Mac

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- Yes. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

**Comment by ye-luo** [comment]
Test this please

File: src/QMCWaveFunctions/BsplineFactory/HybridRepCenterOrbitals.h [hunk 1]
```diff
@@ -398,6 +400,71 @@ class AtomicOrbitals
...
+          const auto cur_elem = i * Nsplines + lidx * Npad + j;
+          ST newval{0.};
+          for (size_t k = 0; k < TrueNOrbs; k++)
+          {
+            const auto index = i * Nsplines + lidx * Npad + k;
+            newval += (*coef_copy_)[index] * rot_mat[k][j];
+          }
+          spl_coefs[cur_elem] = newval;
+        }
+#endif
```
[the remark this answers is not in the document]
Comment (camelto2) [reply src/QMCWaveFunctions/BsplineFactory/HybridRepCenterOrbitals.h:466]: For C2R, it is incorrect. However, we don't have applyRotation implemented at all for SplineC2R, which the hybrid orbital would need in the first place. 

I guess we need a guard of some kind here

**Comment by camelto2** [comment]
@ye-luo There were two offload tests that failed, due to applyRotation not being implemented for SplineC2COMPTarget being used in hybrid rep. I cleaned up the logic for whether or not hybrid orbitals support applyRotation where it checks the value of rotation support in the spline base. 

Note, this should also guard against SplineC2R, since we don't have applyRotation implemented for that yet.

**Review by ye-luo — APPROVED** [review]

**Comment by ye-luo** [comment]
@camelto2 could you open an issue listing what is working with rotation and what is not for the feature matrix C2C, C2R, R2R, offload, hybrid.

**Pull request merged**